wlauto/workloads/multiapp/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,167 @@
+package com.arm.wlauto.uiauto.multiapp;
+
+import android.os.Bundle;
+
+// Import the uiautomator libraries
+import com.android.uiautomator.core.UiObject;
+import com.android.uiautomator.core.UiScrollable;
+import com.android.uiautomator.core.UiSelector;
+
+import com.arm.wlauto.uiauto.UxPerfUiAutomation;
+
+import java.io.File;
+import java.util.concurrent.TimeUnit;
+import java.util.LinkedHashMap;
+
+public class UiAutomation extends UxPerfUiAutomation {
+
+    public static String TAG = "uxperf_multiapp";
+
+    public Bundle parameters;
+    private String outputDir;
+    private LinkedHashMap<String, Timer> timingResults = new LinkedHashMap<String, Timer>();
+
+    public void runUiAutomation() throws Exception {
+        parameters = getParams();
+
+        com.arm.wlauto.uiauto.googlephotos.UiAutomation googlephotos =
+            new com.arm.wlauto.uiauto.googlephotos.UiAutomation();
+
+        confirmAccess();
+        googlephotos.dismissWelcomeView();
+
+        // select the first photo
+        googlephotos.tagPhoto(0);
+        sendToGmail();
+
+        // select the second photo
+        googlephotos.tagPhoto(1);
+        logIntoSkype();
+
+        // Skype won't allow us to login and share on first visit so invoke
+        // once more from googlephotos
+        pressBack();
+        pressBack();
+        googlephotos.tagPhoto(1);
+
+        sendToSkype();
+
+        writeResultsToFile(timingResults, parameters.getString("output_file"));
+    }
+
+    private void sendToGmail() throws Exception {
+
+        final String dumpsysTag = "sendToGmail";
+        final String PACKAGE = "com.google.android.gm";
+        String outputDir = parameters.getString("output_dir", "/sdcard/wa-working");
+
+        com.arm.wlauto.uiauto.gmail.UiAutomation gmail =
```

Comment:
Agreed. I think the timing you are talking about though are captured in the shareUsingApp() method.